### PR TITLE
set tags with help from the compiler

### PR DESCRIPTION
**Pull request by bensigelman**
@slimsag @yurishkuro thoughts?

### CODE REVIEW

File: opentracing/opentracer.go [hunk 1]
```diff
@@ -11,29 +11,22 @@ type OpenTracer interface {
...
 	//
 	//     val tracer OpenTracer = ...
 	//
 	//     sp := tracer.StartTrace("GetFeed")
 	//
-	//     sp := tracer.StartTrace(
-	//         "HandleHTTPRequest",
-	//         "user_agent", req.UserAgent,
-	//         "lucky_number", 42)
+	//     sp := tracer.StartTrace("HandleHTTPRequest").
```
Comment (bensigelman) [inline opentracing/opentracer.go]: I don't feel strongly, but I would weakly vote for merging this PR.

File: opentracing/standardtracer/standardimpl.go [hunk 2]
```diff
@@ -41,6 +41,16 @@ func (s *standardSpan) SetTag(key string, value interface{}) opentracing.Span {
 	return s
 }
 
+func (s *standardSpan) SetTags(tags opentracing.Tags) opentracing.Span {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	for k, v := range tags {
+		s.raw.Tags[k] = fmt.Sprint(v)
```
Comment (yurishkuro) [inline opentracing/standardtracer/standardimpl.go]: @bensigelman  Ben, didn't you originally want to allow repeated tags?

**Comment by yurishkuro** [comment]
lgtm

**Comment by slimsag** [comment]
LGTM -- I do think this is an improvement.

File: opentracing/standardtracer/standardimpl.go (hunk 2 above)
Comment (bensigelman) [inline opentracing/standardtracer/standardimpl.go]: at one point, though I had some cold feet when I started to codify it... felt like it was making the simple case un-simple (and it'll be easier to add repeated tags later than to remove them).

**Comment by yurishkuro** [comment]
Fixes #9

**Comment by adriancole** [comment]
agreed. this is a move forward

**Pull request merged**